File: Reverse a Stack[4-August-2023]/Solution.cpp
//  C++ CODE <<

class Solution{
public:
    void insertAtLast(stack<int> &St,int val){
        if(St.size()==0){
            St.push(val);
            return;
        }
        
            int el=St.top();
            St.pop();
            insertAtLast(St,val);
            St.push(el);
    }
    
    void Reverse(stack<int> &St){
        if(St.size()==0) return;
        
        int el=St.top();
        St.pop();
        
        Reverse(St);
        
        insertAtLast(St,el);
    }
};

File: Reverse a Stack[4-August-2023]/Solution.java
//  >> JAVA CODE <<

class Solution
{ 
    static void insertAtLast(Stack<Integer> s,int val)
    {
        if(s.isEmpty()){
            s.add(val);    
            return;
        }
        
        int el=s.pop();
        insertAtLast(s,val);
        s.add(el);
    }
    static void reverse(Stack<Integer> s)
    {
        if(s.isEmpty()) return;
        
        int el=s.pop();
        
        reverse(s);
        
        insertAtLast(s,el);
    }
}